src/data-access/permission/permission.dao.ts

```diff
@@ -0,0 +1,37 @@
+import { PermissionEntity, PermissionEntityModel } from './permission.entity';
+import { NotFoundError } from '../../models';
+import { PermissionDao } from '..';
+
+export class PermissionDaoImpl implements PermissionDao {
+  getById(id: number): Promise<PermissionEntityModel> {
+    return PermissionEntity.findByPk(id)
+      .then(processNullableEntity)
+      .then((result: PermissionEntityModel) => {
+        return {
+          id: result.id,
+          value: result.value,
+        };
+      });
+  }
+  getByName(name: string): Promise<PermissionEntityModel> {
+    return PermissionEntity.findOne({ where: { value: name } })
+      .then(processNullableEntity)
```

Comment:
Multiple 'then' is not necessary here, because there is no async operations except `PermissionEntity.findOne`. Please collapse it, or even use async/await everywhere:)

Reply:
removed second then statement